gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/login/LoginHideFormHandler.java

```diff
@@ -47,37 +50,59 @@ public LoginHideFormHandler(Domain domain) {
     public void handle(RoutingContext routingContext) {
         final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
         final List<IdentityProvider> socialProviders = routingContext.get(SOCIAL_PROVIDER_CONTEXT_KEY);
-        final LoginSettings loginSettings = LoginSettings.getInstance(domain, client);
-        var optionalSettings = ofNullable(loginSettings).filter(Objects::nonNull);
-        boolean isHideForm = optionalSettings.map(LoginSettings::isHideForm).orElse(false);
+        boolean isHideFormEnabled = ofNullable(LoginSettings.getInstance(domain, client))
+                .map(LoginSettings::isHideForm)
+                .orElse(false);
+        boolean hasExactlyOneSocialProvider = socialProviders != null && socialProviders.size() == 1;
 
-        // hide form option disabled, continue
-        if (!isHideForm) {
-            routingContext.next();
-            return;
-        }
+        boolean shouldSkipLoginForm = isHideFormEnabled && hasExactlyOneSocialProvider;
```

Comment:
question: should it be a OR (||) instead of AND (&&) operator as in previous implementation, there were 2 "if" statements?

Reply:
The "ifs" were independent (no "else" part), and had the same outcome (context.next() + return), so that's equivalent to a single if with OR of all these conditions